### ISSUE TITLE
AGP8 build error

### ISSUE THREAD

**Issue by Meta-Learn-ApS** [bluefireteam/audioplayers#1535]
### Checklist

- [X] I read the [troubleshooting guide](https://github.com/bluefireteam/audioplayers/blob/main/troubleshooting.md) before raising this issue
- [X] I made sure that the issue I am raising doesn't already exist

### Current bug behaviour

I get this error when I try to build:
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':audioplayers_android:compileReleaseKotlin'.
> 'compileReleaseJavaWithJavac' task (current target is 1.8) and 'compileReleaseKotlin' task (current target is 11) jvm target compatibility should be set to the same Java version.
  Consider using JVM toolchain: https://kotl.in/gradle/jvm/toolchain
```

I got the same error for the flutter_timezone package, and fixed it with some changes to the build.gradle file, see https://github.com/tjarvstrand/flutter_timezone/issues/10

However, I do not understand the AGP, so not sure if this is a good solution.

Now the same problem happens with this package, so if anyone can help it would be nice ;)

### Steps to reproduce

1. Execute `flutter run` on the code sample 
2. ... 
3. ... 

### Code sample

<details>
<summary>Code sample</summary>

```dart
void main() {
}
```

</details>

### Affected platforms

Android

### AudioPlayers Version

audioplayers: 4.1.0

### Build mode

release

### Logs

```
my relevant logs
```

<details>
  <summary>Full Logs</summary>

  

  ```
  my full logs or a link to a gist
  ```

  Flutter doctor:
  ```
  Output of: flutter doctor -v
  ```
</details>

### Working on PR

no way

**Comment by bigzhu** [comment]
same as when build apk

**Comment by bigzhu** [comment]
```
-    ext.kotlin_version = '1.7.0'
+    ext.kotlin_version = '1.8.21'
```

**Comment by Gustl22** [comment]
Set the java compatibility target:
https://stackoverflow.com/a/69084853/5164462

Like shown in our example:
https://github.com/bluefireteam/audioplayers/blob/92995357c4e29d63537d66f478dca8f655e99a08/packages/audioplayers/example/android/app/build.gradle#L32

**Comment by dovahkiin98** [comment]
Changing the Java version of the application in `build.gradle` doesn't change anything, it still gives out the same error no matter what you change it to

**Comment by dovahkiin98** [comment]
Think the issue is because the `build.gradle` for the library doesn't define any `java` version. All other packages I've seen manually define it as 1_8

**Comment by dovahkiin98** [comment]
Seems the main issue is using Kotlin 1.8.10 and it's stdlib dependency. If you use kotlin 1.7.10 it works.

Still, that shouldn't be the case. The dev shouldn't be forced to use an older version of Kotlin just to use a library

**Comment by dovahkiin98** [comment]
A different issue rises when upgrading the AGP version to something newer

**Comment by uj** [comment]
Same issue here.  I can't use audioplayers with Kotlin 1.8.22, downgrading to 1.7.10 fixes it.

**Comment by Gustl22** [comment]
Should be solved with #1503, but one needs to wait for the next release :)
